skema/rest/integrated_text_reading_proxy.py

```diff
@@ -0,0 +1,209 @@
+            # Merge both with some de de-duplications
+            params = {"gpt_key": OPENAI_KEY}
+
+            data = {
+                "mit_file": json.dumps(
+                    AttributeCollection(attributes=canonical_mit).json()
+                ),
+                "arizona_file": json.dumps(
+                    AttributeCollection(attributes=canonical_arizona).json()
+                ),
+            }
+            response = requests.post(
+                f"{MIT_TR_ADDRESS}/integration/get_mapping", params=params, data=data
+            )
+
+            if response.status_code == 200:
+                map_data = response.text
+                map_path = os.path.join(tmpdirname, "mapping.txt")
+                with open(map_path, "w") as f:
+                    f.write(map_data)
+                merged_collection = merge_collections(
+                    a_collection=collections[0],
+                    m_collection=collections[1],
+                    map_path=Path(map_path),
+                )
+
+                # Return the merged collection here
+                return merged_collection
+
+    # Merge the collections into a attribute collection
+    attributes = list(it.chain.from_iterable(c.attributes for c in collections))
+
+    return AttributeCollection(attributes=attributes)
+
+
+# End utility code for the endpoints
+
+
+@router.post(
+    "/integrated-text-extractions",
+    summary="Posts one or more plain text documents and annotates with SKEMA and/or MIT text reading pipelines",
+)
+async def integrated_text_extractions(
+    texts: TextReadingInputDocuments,
+    annotate_skema: bool = True,
+    annotate_mit: bool = True,
+) -> TextReadingAnnotationsOutput:
+    texts = texts.texts
+    skema_extractions = [[] for t in texts]
+
+    if annotate_skema:
+        try:
+            skema_extractions = annotate_text_with_skema(texts)
+        except Exception as ex:
+            print(f"Problem annotating with Skema: {ex}")
+
+    mit_extractions = [[] for t in texts]
+
+    if annotate_mit:
+        mit_extractions = annotate_text_with_mit(texts)
+
+    results = list()
+    errors = list()
+    assert len(skema_extractions) == len(
+        mit_extractions
+    ), "Both pipeline results lists should have the same length"
+    for skema, mit in zip(skema_extractions, mit_extractions):
+        if annotate_skema and isinstance(skema, str):
+            errors.append(TextReadingError(pipeline="SKEMA", message=skema))
+
+        if annotate_mit and isinstance(mit, str):
+            errors.append(TextReadingError(pipeline="MIT", message=mit))
+
+        normalized = normalize_extractions(
+            arizona_extractions=skema, mit_extractions=mit
+        )
+        results.append(
+            TextReadingDocumentResults(
+                data=normalized if normalized.attributes else None,
+                errors=errors if errors else None,
+            )
+        )
+    return TextReadingAnnotationsOutput(outputs=results)
```

Comment:
You can change the status code to non-200 if you have any errors in your TexReadingDocumentResults (see https://fastapi.tiangolo.com/advanced/response-change-status-code/#use-a-response-parameter).

Reply:
This is dealt with, by using status code 207